### PR TITLE
feat!: updates for v1

### PR DESCRIPTION
**Pull request by bharathkkb**
Remove creds input, remove SA tests and update related docs

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -55,26 +50,11 @@ async function run(): Promise<void> {
...
-      logWarning(
-        'The "credentials" input is deprecated. ' +
-          'Please switch to using google-github-actions/auth which supports both Workload Identity Federation and JSON Key authentication. ' +
-          'For more details, see https://github.com/google-github-actions/get-gke-credentials#authorization',
-      );
-
-      credentialsJSON = parseCredential(credentials);
-    }
-
     // Pick the best project ID.
```
Comment (sethvargo) [inline src/main.ts:53]: I think we can simplify this logic to `presence(core.getInput('project_id')) ||  presence(clusterName.projectID) || presence(process.env?.GCLOUD_PROJECT)`?

**Review by sethvargo — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -129,7 +123,7 @@ jobs:
 
     steps:
     - id: 'auth'
-      uses: 'google-github-actions/auth@v0'
+      uses: 'google-github-actions/auth@v1'
```
Comment (verbanicm) [inline README.md:126]: Can you do a once over to make sure all references to v0 are updated to v1, including for this action. This is a chicken vs egg problem but we can update in one shot prior to releasing.

File: src/main.ts (hunk 1 above)
Comment (bharathkkb) [reply src/main.ts:53]: We could for project ID and location but I am thinking the verbose logging could be helpful to debug issues?

File: README.md (hunk 2 above)
Comment (bharathkkb) [reply README.md:126]: sg, all updated to v1.

**Review by verbanicm — APPROVED** [review]

**Review by sethvargo — APPROVED** [review]

**Pull request merged**